CMakeLists.txt

```diff
@@ -24,13 +13,10 @@ string(COMPARE EQUAL ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_SOURCE_DIR} PROJECT_IS_
 option(GSL_INSTALL "Generate and install GSL target" ${PROJECT_IS_TOP_LEVEL})
 option(GSL_TEST "Build and perform GSL tests" ${PROJECT_IS_TOP_LEVEL})
 
-# This GSL implementation generally assumes a platform that implements C++14 support.
-set(gsl_min_cxx_standard "14")
-
-if (PROJECT_IS_TOP_LEVEL)
-    gsl_set_default_cxx_standard(${gsl_min_cxx_standard})
-else()
-    gsl_client_set_cxx_standard(${gsl_min_cxx_standard})
+# Because GSL is a header only library, that assumes at minimum C++14 support
```

Comment:
The way I'm reading this comment: the fact that the GSL is a header-only library implies that C++14 is the minimum-supported version. Is that correct? Why does header-only imply a minimum of C++14?

Reply:
Yeah that's worded weirdly. Apologies.

Reply:
I reworded it to `GSL exposes an interface to clients that requires at minimum C++14` which is more concise.

Reply:
>  Why does header-only imply a minimum of C++14?

It doesn't. My wording was awkward.

As per the READMEmd

"The entire implementation is provided inline in the headers under the [gsl](https://github.com/microsoft/GSL/blob/main/include/gsl) directory. The implementation generally assumes a platform that implements C++14 support."

This is enforceable with CMake via `target_compile_features(GSL INTERFACE "cxx_std_14")`

Reply:
The CMake documentation about this scenario is quite nice:
https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html#requiring-language-standards

Reply:
Also CI only tests C++ 14, 17, 20 FWIW

```yml
parameters:
  CXXVersions: [ 14, 17, 20 ]
  buildTypes: [ 'Debug', 'Release' ]
  image: ''
```

Reply:
thanks for clarifying